src/utils/read-file.js

```diff
@@ -0,0 +1,11 @@
+import fs from 'fs';
+import promisify from 'es6-promisify';
+
+var readFile = promisify(fs.readFile);
+
+export default async function (file) {
+    var data = await readFile(file);
+
+    //NOTE: removing byte order mark (BOM)
+    return data.toString().replace(/^\uFEFF/, '');
+};
```

Comment:
Let's move it to the utils inside `compiler` since the whole code from there will be moved to the separate package soon. And let's rename it to `readSourceFile`